### PR TITLE
Issue #14053: remove cycles

### PR DESCRIPTION
**Pull request by romani**
Issue #14053

I do not know how to make it work
I just added few empty lines to
https://github.com/checkstyle/checkstyle/blob/e4942b9158f910f1089ada94f78e4a94e65d3f23/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java#L94
and it test failed.

we have `freeze.refreeze=true` and it regenerate file each time, but it should be `freeze.refreeze=false` , as "true"  should be used in case to allow new vioaltions.

but even I did this, freeze file generation is not stable, I increased number of nodes and file become different, ok, I added few empty lines and some cycles are removed from file, even I did nothing.

I think something is not working for this Rule and freeze file, Freeze might work for other rules, but cycle+freeze is weird.

-------

We might need to create issue to resolve cycles and at the end enable this Rule.
One more idea is to write our own Store Persisting - https://www.archunit.org/userguide/html/000_Index.html#_violation_store
to keep only fact of cycle:
```
✔ ~/java/github/romani/checkstyle [master|⚑ 2] 
$ ag --nofilename -A 5 "Cycle detected:" | grep -vE "Dependencies|Method|Const"
Cycle detected: Slice api -> \
                Slice utils -> \
                Slice api\

Cycle detected: Slice api -> \
                Slice utils -> \
                Slice checks.javadoc -> \
                Slice api\

Cycle detected: Slice api -> \
                Slice utils -> \
                Slice checks.javadoc -> \
                Slice checks.naming -> \
                Slice api\

Cycle detected: Slice api -> \
                Slice utils -> \
                Slice checks.javadoc.utils -> \
                Slice api\

Cycle detected: Slice api -> \
                Slice utils -> \
                Slice checks.naming -> \
                Slice api\

Cycle detected: Slice api -> \
                Slice utils -> \
                Slice xpath -> \
                Slice api\

Cycle detected: Slice checks.javadoc -> \
                Slice checks.naming -> \
                Slice utils -> \
                Slice checks.javadoc\

Cycle detected: Slice checks.javadoc -> \
                Slice utils -> \
                Slice checks.javadoc\

Cycle detected: Slice checks.naming -> \
                Slice utils -> \
                Slice checks.naming\

Cycle detected: Slice utils -> \
                Slice xpath -> \
                Slice utils\
```

### CODE REVIEW

**Comment by romani** [comment]
while we have implementations in our api folders, we can not fix:
```
Cycle detected: Slice api -> \
                Slice utils -> \
                Slice ....... -> \
                Slice api\
```

File: src/test/resources/archunit.properties
Comment (nrmancuso) [inline src/test/resources/archunit.properties:1]: Is it possible that we are hitting some of the max numbers in this file, and that is why test isn't functioning as expected?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/resources/archunit.properties
Comment (romani) [reply src/test/resources/archunit.properties:1]: Maybe, but made multiplied numbers by million, still a problem.

**Comment by romani** [comment]
The biggest problem is facts at https://github.com/checkstyle/checkstyle/pull/14060#issuecomment-1826795578
We can not  resolve cycles until we have actual implementation in our API package. Moving all to interfaces is not easy approach, as it will trigger heated debates.
Issue to arch project will look like "how to activate this in our big project", it would still be good to add it from scratch,  to not involve them in confusing history.

Referenced PR should be we add test class and store in first commit, second commit we add new empty lines and we have some issue. We add new cycle, violation is weird.

But still better to remove and start from scratch. Test is minimalistic.

**Review by nrmancuso — APPROVED** [review]
Ok, I really don't agree with the need to do all this extra work before we report some issue to ArchUnit, but let's not waste any more time on this.

**Pull request merged**